collect_app/src/main/java/org/odk/collect/android/external/FormUriActivity.kt

```diff
@@ -28,26 +28,37 @@ class FormUriActivity : Activity() {
         DaggerUtils.getComponent(this).inject(this)
         setTheme(ThemeUtils(this).appTheme)
 
-        val firstProject = projectsRepository.getAll().first()
-        val uri = intent.data
-        val uriProjectId = uri?.getQueryParameter("projectId")
-        val projectId = uriProjectId ?: firstProject.uuid
-
-        logAnalytics(uriProjectId)
-
-        if (projectId == currentProjectProvider.getCurrentProject().uuid) {
-            startActivity(
-                Intent(this, FormEntryActivity::class.java).also {
-                    it.data = uri
-                    intent.extras?.let { sourceExtras -> it.putExtras(sourceExtras) }
-                }
-            )
-        } else {
+        val projects = projectsRepository.getAll()
+        if (projects.isEmpty()) {
             AlertDialog.Builder(this)
-                .setMessage(R.string.wrong_project_selected_for_form)
+                .setMessage(R.string.no_projects_detected)
                 .setPositiveButton(R.string.ok) { _, _ -> finish() }
                 .create()
                 .show()
+        } else {
+            val firstProject = projects.first()
+            val uri = intent.data
+            val uriProjectId = uri?.getQueryParameter("projectId")
+            val projectId = uriProjectId ?: firstProject.uuid
+
+            logAnalytics(uriProjectId)
+
+            if (projectId == currentProjectProvider.getCurrentProject().uuid) {
+                startActivity(
+                    Intent(this, FormEntryActivity::class.java).also {
+                        it.data = uri
+                        intent.extras?.let { sourceExtras -> it.putExtras(sourceExtras) }
+                    }
+                )
+            } else {
+                AlertDialog.Builder(this)
```

Comment:
Dialogs should use `MaterialAlertDialogBuilder` as per #4914.

Reply:
Fixed.